metrics/sources/kubelet/kubelet.go

```diff
@@ -132,25 +157,12 @@ func (this *kubeletMetricsSource) decodeMetrics(c *cadvisor.ContainerInfo) (stri
 			}
 		}
 
+		// No Kubernetes metadata so treat this as a system container.
 		if cName == "" || ns == "" || podName == "" {
```

Comment:
Do we have to support Kubernetes 1.0.x btw? We could get rid of this bit of logic & rely on Docker labels if not.

Having said that, I don't see any chance of failure in extracting the k8s container name from the Docker container name. This is what is done in the logging component successfully. Pod name is also present in the Docker container name, as is namespace - we can extract all that info from the Docker container name, but as I said above I'd prefer to drop support for 1.0.x & rely on Docker labels in 1.1.x.